src/lib/kafka/KafkaConnectionManager.cpp

```diff
@@ -0,0 +1,445 @@
+  {
+    int kafka_flush_timeout_ms = KAFKA_FLUSH_TIMEOUT_MS;
+    rd_kafka_resp_err_t err = rd_kafka_flush(producer, kafka_flush_timeout_ms); // Wait KAFKA_FLUSH_TIMEOUT_MS seconds to send pending messages
+    if (err != RD_KAFKA_RESP_ERR_NO_ERROR)
+    {
+      LM_E(("Error al hacer flush de mensajes para %s: %s",
+           endpoint.c_str(), rd_kafka_err2str(err)));
+    }
+    rd_kafka_destroy(producer);     // Free resources (equivalent to mosquitto_destroy)
+  }
+  //FIXME #4666: Implement alarm feature for Kafka notifications
+  //alarmMgr.kafkaConnectionError(endpoint, "disconnected");
+}
+
+/* ****************************************************************************
+*
+* KafkaConnectionManager::getConnection -
+*/
+KafkaConnection* KafkaConnectionManager::getConnection(const std::string& endpoint)
+{
+  // Kafka uses endpoint "broker1:9092,broker2:9092"
+
+  if (connections.find(endpoint) == connections.end())
+  {
+    LM_T(LmtKafkaNotif, ("Starting a new KAFKA broker connection for %s", endpoint.c_str()));
+
+    KafkaConnection* kConn = new KafkaConnection();
+    kConn->endpoint = endpoint;
+
+    // Basic configuration (as in MQTT)
+    rd_kafka_conf_t* conf = rd_kafka_conf_new();
+    if (rd_kafka_conf_set(conf, "bootstrap.servers", endpoint.c_str(), NULL, 0) != RD_KAFKA_CONF_OK)
+    {
+      LM_E(("Runtime Error (Kafka config failed for %s)", endpoint.c_str()));
+      delete kConn;
+      return NULL;
+    }
+    rd_kafka_conf_set_opaque(conf, kConn);
+
+    // Force UTF-8 and disable conversions
+    rd_kafka_conf_set(conf, "message.encoding", "utf-8", NULL, 0);
+
+    // Buffer configuration
+    rd_kafka_conf_set(conf, "queue.buffering.max.ms", KAFKA_MAX_BUFFERING_MS_STR, NULL, 0);
+    rd_kafka_conf_set(conf, "message.max.bytes", KAFKA_MAX_MESSAGE_BYTES_STR, NULL, 0);
+
+    // Connection callback (similar to MQTT)
+    rd_kafka_conf_set_dr_msg_cb(conf, [](rd_kafka_t* rk, const rd_kafka_message_t* msg, void* opaque)
+    {
+      KafkaConnection* kConn = static_cast<KafkaConnection*>(opaque);
+
+      if (msg->err == RD_KAFKA_RESP_ERR_NO_ERROR)
+      {
+        LM_T(LmtKafkaNotif, ("Kafka notification successfully published at %s on topic %s",
+               kConn->endpoint.c_str(),
+               rd_kafka_topic_name(msg->rkt)));
+      }
+      else
+      {
+        LM_E(("Kafka delivery failed at %s: %s",
+          kConn->endpoint.c_str(),
+          rd_kafka_err2str(msg->err)));
+      }
+
+      kConn->connectionCallbackCalled = true;
+      sem_post(&kConn->connectionSem);
+    });
+
+
+    // Create producer
+    kConn->producer = rd_kafka_new(RD_KAFKA_PRODUCER, conf, NULL, 0);
+    if (!kConn->producer)
+    {
+      LM_E(("Runtime Error (Failed to create Kafka producer for %s)", endpoint.c_str()));
+      delete kConn;
+      return NULL;
+    }
+
+    // Wait for connection
+    sem_init(&kConn->connectionSem, 0, 0);
+    kConn->connectionCallbackCalled = false;
+
+    rd_kafka_poll(kConn->producer, 0);
```

Comment:
Really needed? Check

Reply:
Not needed at the end. Removed in [7a1e653](https://github.com/telefonicaid/fiware-orion/pull/4700/commits/7a1e653f01a8768771640b42631da430c9de86de)